File: 21.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        
        h = ListNode(-1)
        p = h
        while (l1!=None or l2!=None):
            if !l1 or (l2 and l2.val < l1.val):
                p.next = l2
                p = p.next
                l2 = l2.next
            else:
                p.next = l1
                p = p.next
                l1 = l1.next

        return h.next
                
                

File: 21.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* p = l1;
        ListNode* q = l2;
        
        ListNode* h = new ListNode(0);
        ListNode* t = h;
        
        while (p!=nullptr or q!=nullptr){
            if (q==nullptr or (p!=nullptr and p->val < q->val) ){
                t->next = p;
                t = p;
                p = p->next;
            }
            else {
                t->next = q;
                t = q;
                q = q->next;
            }
        }
        
        return h->next;
        
        
        
    }
};